src/main/java/org/folio/circulation/domain/policy/LoanPolicy.java

```diff
@@ -65,19 +65,19 @@ public static LoanPolicy from(JsonObject representation) {
 
   public Result<Loan> renew(Loan loan, DateTime systemDate) {
     //TODO: Create HttpResult wrapper that traps exceptions
+    List<ValidationError> errors = new ArrayList<>();
     try {
       if (isNotLoanable()) {
-        return failedValidation(errorForPolicy("item is not loanable"));
-      }
-      if(isNotRenewable()) {
-        return failedValidation(errorForPolicy("loan is not renewable"));
+        errors.add(errorForPolicy("item is not loanable"));
```

Comment:
Previously, the renewal failed quickly when the loan policy states that the item is not loanable.

According to the UI, if an item isn't loanable, it also isn't renewable.

When an item isn't loanable, there should be no loans policy, which means that it isn't possible to calculate a due date.

What errors are we expecting to accumulate with this?

Reply:
Yes, I agree with you, I should immediately return the error if isNotLoanable() = true